packages/backend-common/src/util/docker.test.ts

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Docker from 'dockerode';
+import mockFs from 'mock-fs';
+import os from 'os';
+import path from 'path';
+import Stream, { PassThrough } from 'stream';
+import { runDockerContainer, UserOptions } from './docker';
+
+const mockDocker = new Docker() as jest.Mocked<Docker>;
+const rootDir = os.platform() === 'win32' ? 'C:\\rootDir' : '/rootDir';
+
+describe('runDockerContainer', () => {
+  beforeEach(() => {
+    mockFs({
+      [rootDir]: {
+        input: mockFs.directory(),
+        output: mockFs.directory(),
+      },
+    });
+
+    jest.spyOn(mockDocker, 'pull').mockImplementation((async (
+      _image: string,
+      _something: any,
+      handler: (err: Error | undefined, stream: PassThrough) => void,
+    ) => {
+      const mockStream = new PassThrough();
+      handler(undefined, mockStream);
+      mockStream.end();
+    }) as any);
+
+    jest
+      .spyOn(mockDocker, 'run')
+      .mockResolvedValue([{ Error: null, StatusCode: 0 }]);
+
+    jest
+      .spyOn(mockDocker, 'ping')
+      .mockResolvedValue(Buffer.from('OK', 'utf-8'));
+  });
+
+  afterEach(() => {
+    mockFs.restore();
+  });
+
+  const imageName = 'dockerOrg/image';
+  const args = ['bash', '-c', 'echo test'];
+  const mountDirs = new Map([
+    [path.join(rootDir, 'input'), '/input'],
+    [path.join(rootDir, 'output'), '/output'],
+  ]);
+  const workingDir = path.join(rootDir, 'input');
+  const envVars = ['HOME=/tmp', 'LOG_LEVEL=debug'];
```

Comment:
A nit, but I'd guess this inclusion infers that this function can only be used if firing up Linux-based Docker containers?  If so, may be useful maybe to throw a comment regarding that to the jsdoc at the beginning of this method. but i'm not otherwise concerned, just noticed!

Reply:
And sorry I should add, I see /tmp used in many places.  So it's sort of, a simple command which has broader implications but there are likely numerous places in here where absolutely it would be only a Linux-based image and i wouldn't know where :)

Reply:
The `runDockerContainer` function could be used to fire up _any_ Docker container with _any_ environment variables. :)

The `HOME=/tmp` is especially helpful for the Create React App templater which runs a Nodejs docker container based on Linux. So, it's good to write a test for that.